scripts/Diffraction/isis_powder/polaris_routines/polaris_algs.py

```diff
@@ -0,0 +1,142 @@
+from __future__ import (absolute_import, division, print_function)
+
+import mantid.simpleapi as mantid
+import os
+from isis_powder.routines import common, yaml_parser
+from isis_powder.routines.RunDetails import RunDetails
+from isis_powder.polaris_routines import polaris_advanced_config
+
+
+def calculate_absorb_corrections(ws_to_correct):
+    mantid.MaskDetectors(ws_to_correct, SpectraList=list(range(0, 55)))
+
+    absorb_dict = polaris_advanced_config.absorption_correction_params
+    geometry_json = {'Shape': 'Cylinder', 'Height': absorb_dict["cylinder_sample_height"],
+                     'Radius': absorb_dict["cylinder_sample_radius"], 'Center': absorb_dict["cylinder_position"]}
+    material_json = {'AttenuationXSection': absorb_dict["attenuation_cross_section"],
+                     'ChemicalFormula': absorb_dict["chemical_formula"],
+                     'ScatteringXSection':  absorb_dict["scattering_cross_section"],
+                     'SampleNumberDensity': absorb_dict["sample_number_density"]}
+
+    mantid.SetSample(InputWorkspace=ws_to_correct, Geometry=geometry_json, Material=material_json)
+
+    ws_to_correct = mantid.ConvertUnits(InputWorkspace=ws_to_correct, OutputWorkspace=ws_to_correct, Target="TOF")
+    ws_to_correct = mantid.MayersSampleCorrection(InputWorkspace=ws_to_correct, OutputWorkspace=ws_to_correct)
+    ws_to_correct = mantid.ConvertUnits(InputWorkspace=ws_to_correct, OutputWorkspace=ws_to_correct, Target="dSpacing")
+
+    return ws_to_correct
+
+
+def get_run_details(run_number, inst_settings):
+    yaml_dict = yaml_parser.get_run_dictionary(run_number=run_number, file_path=inst_settings.cal_mapping_file)
+
+    if inst_settings.chopper_on:
+        chopper_config = yaml_dict["chopper_on"]
+    else:
+        chopper_config = yaml_dict["chopper_off"]
+
+    label = yaml_dict["label"]
+    empty_runs = chopper_config["empty_run_numbers"]
+    vanadium_runs = chopper_config["vanadium_run_numbers"]
+
+    splined_vanadium_name = _generate_splined_van_name(chopper_on=inst_settings.chopper_on,
+                                                       vanadium_run_string=vanadium_runs)
+
+    grouping_full_path = os.path.join(inst_settings.calibration_dir, inst_settings.grouping_file_name)
```

Comment:
Same note as on the previous path comment, consider using expand and abspath if you think it's possible someone would pass `~` or env variables